### PR TITLE
Backend/verify signup in out

### PR DESCRIPTION
**Pull request by notiona**
## Description

This pull request implements all the Auth related APIs(signup, signin, signout, verify) along with the verification email sending/validating process.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by JadenJSPark — APPROVED** [review]
ddabong

File: backend/linklink/views.py [hunk 1]
```diff
@@ -149,35 +237,35 @@ def signout(request):
...
-    onechon_list = set(onechon_list)
-    onechon_list.remove(linklinkuser)
-    # exclude exclude_linklinkuser
-    if exclude_linklinkuser is not None:
-        onechon_list.remove(exclude_linklinkuser)
-    return list(onechon_list)
+    # Find Verification object with token=token
+    verification_found = get_object_or_404(Verification, token=token)
+    # Register: check token expire, set user's emailValidated as True
+    if verification_found.purpose == "Register": # pragma: no branch
```
Comment (hunkim98) [inline backend/linklink/views.py:252]: enum을 사용해 verification purpose구현해도 좋을 것 같네요

**Review by hunkim98 — COMMENTED** [review]
고생하셨습니다~

File: backend/linklink/views.py (hunk 1 above)
Comment (notiona) [reply backend/linklink/views.py:252]: 모델 상에서 enum으로 구현되어 있어서 여기서는 따로 적용하지 않겠습니다

File: backend/backend/settings.py [hunk 2]
```diff
@@ -137,3 +137,6 @@
 
 # Homepage Base Url
 HOMEPAGE_URL = "http://127.0.0.1:8000/"
```
[the remark this answers is not in the document]
Comment (notiona) [reply backend/backend/settings.py]: 감사합니다. 적용했습니다!

**Pull request merged**